### PR TITLE
Add more boolean operators for booleans and strings

### LINKED ISSUE #16: Can't perform binary operations on null
Trying to evaluate `var == null` when `var` is not defined in the context leads to the following error result value:

`Invalid binary operation, left: null, right: null, operation: Equal`

### LINKED ISSUE #17: Can't perform NotEqual operation on two strings
Trying to evaluate `"a" != "b"`:

`Invalid binary operation, left: "a", right: "b", operation: Not equal`

### LINKED ISSUE #22: Support for more `boolean` binary expressions
Similar to #15, binary expressions with boolean values are not available. AND and OR are supported, but 

`==`
`!=`
XOR

…and others are missing.

Without a boolean comparison (`==`) or NOT (`!`) this makes boolean context values much less useful.

### PR DESCRIPTION
**Pull request by jhugman**
This PR adds more comprehensive coverage around equality testing (`==` and `!=`).

This gives us comparisons between booleans, nulls and strings; not equal is supported in terms of equality.

Fixes #22 (boolean equals and not equals)
Fixes #16 (null equals and not equals)
Fixes #17 (string not equals).

As part of tidy up of booleans, `<`, `<=`, `>=`, and `>` are all implemented for Strings.

### CODE REVIEW

File: jexl-eval/src/lib.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ impl Truthy for Value {
     fn is_truthy(&self) -> bool {
         match self {
             Value::Bool(b) => *b,
-            Value::Null => true,
+            Value::Null => false,
```
Comment (jhugman) [inline jexl-eval/src/lib.rs:60]: Bug fix.

File: jexl-eval/src/lib.rs [hunk 2]
```diff
@@ -280,18 +289,45 @@ impl<'a> Evaluator<'a> {
                     OpCode::LessEqual => value!(left <= right),
                     OpCode::GreaterEqual => value!(left >= right),
                     OpCode::Equal => value!((left - right).abs() < EPSILON),
-                    OpCode::NotEqual => value!((left - right).abs() > EPSILON),
+                    OpCode::NotEqual => value!((left - right).abs() >= EPSILON),
```
Comment (jhugman) [inline jexl-eval/src/lib.rs:292]: Handles the rare case when left - right was exactly EPSILON.

File: jexl-eval/src/lib.rs [hunk 3]
```diff
@@ -651,4 +687,131 @@ mod tests {
             value!([{"bobo": 50, "fofo": 100}, {"bobo": 60, "baz": 90}])
         );
     }
+
+    #[test]
+    fn test_binary_equals_and_not_equals() {
```
Comment (travis79) [inline jexl-eval/src/lib.rs:692]: I'm curious about your choice of the word `binary` for this and the next test that you added. Is it a reference to the argument count for what is being tested (i.e. testing _two_ arguments for equality/inequality)?  Just a very uber nit, but the word choice made me assume there were going to be some bitwise magic happening

**Review by travis79 — APPROVED** [review]
LGTM! Nice catch on the truthiness of `null`!

File: jexl-eval/src/lib.rs (hunk 3 above)
Comment (jhugman) [reply jexl-eval/src/lib.rs:692]: These are operators with two operands.

Note to self: we need to implement unary operators `-`, `+` and `!`.

Comment (travis79) [reply jexl-eval/src/lib.rs:692]: Ahh, that makes sense now!

**Pull request merged**